tests/citeproc.rs

```diff
@@ -597,6 +620,153 @@ where
+
+    pub(super) fn render(
+        bib: &hayagriva::RenderedBibliography,
+        output: &mut String,
+    ) -> Result<(), fmt::Error> {
+        output.push_str(r#"<div class="csl-bib-body">"#);
+        for item in &bib.items {
+            // TODO: Verify whether this automatically implies left-margin
+            // followed by right-inline (cf. test bugreports_AsmJournals.txt)
+            render_item(item, output)?;
+        }
+        output.push_str("</div>");
+        Ok(())
+    }
+
+    fn render_item(
+        item: &hayagriva::BibliographyItem,
+        output: &mut String,
+    ) -> Result<(), fmt::Error> {
+        output.push_str(r#"<div class="csl-entry">"#);
+        if let Some(field) = &item.first_field {
+            render_child(field, output)?;
+        }
+        for child in &item.content.0 {
+            render_child(child, output)?;
+        }
+        output.push_str("</div>");
+        Ok(())
+    }
+
+    fn render_child(child: &ElemChild, output: &mut String) -> Result<(), fmt::Error> {
+        match child {
+            ElemChild::Text(formatted) => render_formatted_text(formatted, output),
+            ElemChild::Elem(e) => render_elem(e, output),
+            elem => elem.write_buf(output, BufWriteFormat::Html),
+        }
+    }
+
+    fn render_formatted_text(
+        text: &hayagriva::Formatted,
+        output: &mut String,
+    ) -> Result<(), fmt::Error> {
+        let formatting = text.formatting;
+        if formatting == Formatting::default() {
+            output.push_str(&text.text);
+            return Ok(());
+        }
+
+        // TODO: Spaces between multiple CSS? (No example with multiple in tests)
+        let mut css = String::new();
+        let mut suffix = String::new();
+        let mut push_elem = |start, end| {
+            output.push_str(start);
+            suffix.insert_str(0, end);
+        };
+
+        match formatting.vertical_align {
+            VerticalAlign::Sub => {
+                push_elem("<sub>", "</sub>");
+            }
+            VerticalAlign::Sup => {
+                push_elem("<sup>", "</sup>");
+            }
+            VerticalAlign::Baseline => {
+                // TODO: Figure out what happens when baseline is combined with
+                // something else.
+                push_elem(r#"<span style="baseline">"#, "</span>");
+            }
+            VerticalAlign::None => {}
+        }
+
+        match formatting.font_weight {
+            FontWeight::Bold => {
+                push_elem("<b>", "</b>");
+            }
+            FontWeight::Light => {
+                // TODO: Find an example test where this is used
+                css.push_str("font-weight:lighter;");
+            }
+            FontWeight::Normal => {
+                // TODO: Do we support <span class="nodecor">...</span>?
+                // It should force 'font-style:normal' when inside bold/italics
+                // Relevant test: flipflop_ItalicsWithOk.txt
```

Comment:
I'd say it's out of scope, especially given that CSL-JSON is not an input format we currently expose to our users.